cloudify_cli/commands/replace_certificates.py

```diff
@@ -0,0 +1,273 @@
+        {'manager': {'new_ca_cert_path': '',
+                     'new_external_ca_cert_path': '',
+                     'new_ldap_ca_cert_path': '',
+                     'cluster_members': []
+                     },
+         'postgresql_server': {'new_ca_cert_path': '',
+                               'cluster_members': []
+                               },
+         'rabbitmq': {'new_ca_cert_path': '',
+                      'cluster_members': []
+                      }
+         }
+    )
+
+
+def validate_config_dict(config_dict, force, logger):
+    errors_list = []
+    _validate_instances(errors_list, config_dict, force, logger)
+    _check_path(errors_list, config_dict['manager']['new_ldap_ca_cert_path'])
+    return errors_list
+
+
+def _validate_username_and_private_key():
+    # TODO: what if the client is also the manager in the AIO case?
+    if (not env.profile.ssh_user) or (not env.profile.ssh_key):
+        raise CloudifyCliError('Please configure the profile ssh-key and '
+                               'ssh-user using the `cfy profiles set` command')
+
+
+def _validate_instances(errors_list, config_dict, force, logger):
+    for instance in 'postgresql_server', 'rabbitmq':
+        _validate_cert_and_key(errors_list,
+                               config_dict[instance]['cluster_members'])
+        _validate_new_ca_cert(errors_list, config_dict, instance, force,
+                              logger)
+
+    _validate_manager_cert_and_key(errors_list,
+                                   config_dict['manager']['cluster_members'])
+    _validate_new_manager_ca_certs(errors_list, config_dict, force, logger)
+
+
+def _validate_new_ca_cert(errors_list, config_dict, instance_name, force,
+                          logger):
+    _validate_ca_cert(errors_list, config_dict[instance_name], instance_name,
+                      'new_ca_cert_path', 'new_cert_path',
+                      config_dict[instance_name]['cluster_members'],
+                      force, logger)
+
+
+def _validate_new_manager_ca_certs(errors_list, config_dict, force, logger):
+    _validate_ca_cert(errors_list, config_dict['manager'], 'manager',
+                      'new_ca_cert_path', 'new_internal_cert_path',
+                      config_dict['manager']['cluster_members'],
+                      force, logger)
+    _validate_ca_cert(errors_list, config_dict['manager'],
+                      'manager', 'new_external_ca_cert_path',
+                      'new_external_cert_path',
+                      config_dict['manager']['cluster_members'],
+                      force, logger)
+    _validate_ca_cert(errors_list,  config_dict['postgresql_server'],
+                      'postgresql_server', 'new_ca_cert_path',
+                      'new_postgresql_client_cert_path',
+                      config_dict['manager']['cluster_members'],
+                      force, logger)
+
+
+def _validate_ca_cert(errors_list, instance, instance_name, new_ca_cert_name,
```

Comment:
it's always a good idea, but especially this function could use a docstring saying what is it actually validating
it seems that this checks that if you change the CA, you must change the corresponding cert for all nodes?

Reply:
Correct, added a docstring